### PR TITLE
Only store menu in appState on Windows

### PR DESCRIPTION
**Pull request by bsclifton**
... and even then, store only the data (don't serialize functions, etc).

Auditors: @bridiver, @bbondy

Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request [as needed](https://github.com/brave/browser-laptop/wiki/Pull-request-process).

Reviewer Checklist:

Tests

- [ ] Adequate test coverage exists to prevent regressions
- [ ] Tests should be independent and work correctly when run individually or as a suite [ref](https://github.com/brave/browser-laptop/wiki/Code-Guidelines#test-dependencies)
- [ ] New files have MPL2 license header

### CODE REVIEW

File: app/browser/menu.js [hunk 1]
```diff
@@ -596,7 +597,10 @@ const createMenu = (state) => {
     })
   }
 
-  appActions.setMenubarTemplate(Immutable.fromJS(template))
+  if (isWindows) {
+    const menuDataOnly = JSON.parse(JSON.stringify(template))
+    appActions.setMenubarTemplate(Immutable.fromJS(menuDataOnly))
+  }
```
Comment (bsclifton) [inline app/browser/menu.js]: This is not ideal- could use some help finding a better way to do this

File: app/common/lib/platformUtil.js [hunk 2]
```diff
@@ -35,12 +35,12 @@ module.exports.getPathFromFileURI = (fileURI) => {
...
 module.exports.isDarwin = () => {
   return process.platform === 'darwin' ||
-    (navigator && navigator.platform === 'MacIntel')
+    (typeof navigator !== 'undefined' && navigator.platform === 'MacIntel')
 }
 
 module.exports.isWindows = () => {
   return process.platform === 'win32' ||
-    (navigator && navigator.platform === 'Win32')
+    (typeof navigator !== 'undefined' && navigator.platform === 'Win32')
```
Comment (bsclifton) [inline app/common/lib/platformUtil.js]: This is needed because `navigator` is undefined in the browser process (which causes a crash).

**Comment by bsclifton** [comment]
I can make an issue for this too (if it makes sense)

**Comment by bsclifton** [comment]
OK made some tweaks, ready for re-review @bridiver @bbondy 

Here are some important notes:
- This PR updates to only store menu in appState on Windows
  - methods are stripped out (will no longer be serialized)
  - with a few thousand bookmarks, the initial stringify/parse takes ~60ms and is never done again
  - menu (if present) is now deleted when saving session

Some notes about existing behavior
The menu is modified in memory when:
  - tab is closed (history menu rebuilt - serialization added w/ this PR taking 0 ms)
  - menu item goes from checked to unchecked (or similar)

**Comment by bbondy** [comment]
Could you post or link up an issue for this? Thanks.

**Comment by bsclifton** [comment]
@bbondy issue created! 😄  (and commit updated to use closing keyword)
https://github.com/brave/browser-laptop/issues/10016

File: app/browser/menu.js [hunk 3]
```diff
@@ -602,7 +606,10 @@ const createMenu = (state) => {
     })
   }
 
-  appActions.setMenubarTemplate(Immutable.fromJS(template))
+  if (isWindows) {
+    const menuTemplate = JSON.parse(JSON.stringify(template))
```
Comment (bbondy) [inline app/browser/menu.js]: I talked to Clifton, the handlers aren't actually used in app state and this is only called once.

**Pull request merged**